### PR TITLE
Make LHLO behave like EHLO

### LINKED ISSUE #123: LMTP server fails to list any extended services
The LHLO service awaits smtp_HELO instead of smtp_EHLO, preventing the greeting from displaying the service menu.

This is a violation of the LMTP spec, RFC 2033.

Patch attached.  This affects https://gitlab.com/mailman/mailman/issues/365

[lmtp_lhlo.zip](https://github.com/aio-libs/aiosmtpd/files/1132381/lmtp_lhlo.zip)

### LINKED ISSUE #124: LMTP server doesn't support require_starttls; require_starttls not RFC-compliant
`smtp.py`'s _handle_client contains the following code:

````Python
   270                  if (self.require_starttls
   271                          and not self._tls_protocol
   272                          and command not in ['EHLO', 'STARTTLS', 'QUIT']):
   273                      # RFC3207 part 4
   274                      await self.push('530 Must issue a STARTTLS command first')
   275                      continue
````

When subclassed for the LMTP server, 'EHLO' needs to be 'LHLO'.

I'm not enough of a Python person to code the change... (It needs an 'unrestricted commands' list that the subclass can override.)

This bug prevents enabling require_starttls for LMTP in Mailman -- https://gitlab.com/mailman/mailman/issues/365

In addition, RFC3207 section 4 (3 paragraphs from the end) requires that NOOP  also be accepted before TLS is established.  This change, of course, is trivial...

### PR DESCRIPTION
**Pull request by emersion**
The current logic makes LHLO behave like the legacy HELO. However the
[RFC says:](https://tools.ietf.org/html/rfc2033#section-4.1)

> The LHLO command has identical semantics to the EHLO command of ESMTP.

Clients connecting to the LMTP server expect extensions such as 8BITMIME
to be advertised. However aiosmtpd won't send any extension string.

Fix this by making LHLO an alias for EHLO instead of HELO.

## Related issue number

* Closes #123
* Closes #124  

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] tox testenvs have been executed in the following environments:
  - [x] Linux (Arch): `{py38}-{nocov,cov,diffcov}, qa, docs`
- [x] Documentation reflects the changes
- [x] Add a news fragment into the `NEWS.rst` file

### CODE REVIEW

**Comment by ddevault** [comment]
Hey, I'm the CEO of SourceHut, and @emersion wrote this patch under our direction.

We are opposed in principle to signing contributor license agreements, regardless of the apparently banality of the CLA in question. The changes @emersion made are available under the same terms as the aiosmtpd project itself: Apache 2.0. This gives you sufficient legal standing to apply the changes without additional fanfare. We use aiosmtpd to process tens of thousands of emails per year, and we're prepared to fork it if you insist.

Alternatively, you could consider adopting something like git and Linux's signed-off-by header, which indicates agreement with the Developer Certificate of Origin.

**Comment by asvetlov** [comment]
@ddevault I have a question to you. Did you found the CLA requirement for https://github.com/python/cpython also uncomfortable and not acceptable or agree to make an exception for that particular project?

**Comment by ddevault** [comment]
I don't think I have signed Python's CLA, and I probably would not do so. However, my level of investment in Python dwarfs that in aiosmtpd by scores, and it's much harder to fork it if I needed to. I have refused to sign Go's CLA, by the way.

**Comment by asvetlov** [comment]
That's sad to hear that CLA frustrates you.
I'm not a lawyer, my thought is the following:
If big projects like Python and Django think that they need to ask for CLA -- maybe aiohttp needs it too.
If aiohttp needs a CLA -- maybe the CLA requirement for all projects from aio-libs is a good idea.

**Comment by ddevault** [comment]
And if the largest project in the world, Linux, doesn't need one - maybe they have it right? The DCO is a good compromise that will get you everything you need to be confident that you can merge the patches.

And what of the hundreds of projects which don't have a CLA? Have you ever heard of any of them regretting it, for reasons other than the leadership's personal greed? It's precisely the horror stories of the fates of projects which have required a CLA, like MongoDB, RedisLabs, and many more, which should discredit the idea of CLAs. You should not sign one, and you *definitely* should not ask for one. Frankly, it's rude to.

**Comment by warsaw** [comment]
I hate to wade into this discussion, but I guess I will for whatever it's worth (probably not much :D).

I've obviously signed the Python CLA, and for Python I think it's both a good idea and very low friction for contributors.  Python has paid legal opinions that require this, and I have zero problem signing it.  Where I've needed my employer to give me permission, none have ever balked.

I've also contributed to and led GNU projects.  On Mailman we required copyright assignment to the FSF for all contributions above a certain size.  Some people didn't like that, but for the most part, it was never a problem.  Also pretty much a legal requirement for contributions.

Both of these are big projects, and both the PSF and FSF have made it quite clear that contributions under the CLA are guaranteed to be released open source / free software.  So there's no question of some nefarious organization profiting off labor of volunteers.  The CLAs are there to protect the legal interests of the organization owning the IP, and since both the PSF and FSF have strongly open reputations, everyone is cool with it.

I have also worked on non-GNU, GPL'd projects owned by other corporations, and there I think the CLAs have been a hinderance.  The biggest problem there was that the CLAs essentially said, "we'll release your contribution as FLOSS, but we reserve the right to include your contribution in closed commercial products".  It's no surprise to me that that stance isn't very well received.

That all said, what is the purpose of a CLA for aio-libs?  Are you afraid of being sued for copyright infringement?  E.g. someone who doesn't sign the CLA contributes something significant, but either they don't own the rights to that, or they are using that as a weapon to shut the project down.  I personally think that's very unlikely in practice, and also not very practical given that for the size of a project like aio-libs, almost all contributions are derivative works anyway, so it's unlikely that such a copyright claim would stand in court.  And it's also pretty unlikely that CLA or no, bad copyright actors could still make your life uncomfortable.

**Comment by asvetlov** [comment]
Thank you @warsaw for the comprehensive comment.
You've convinced me that the CLA is not *mandatory* and aio-libs can just drop it.

I apologize for the inconvenience, I never had a target to make harm but was trying to satisfy all really required things.

Will drop the CLA in the moment.

**Comment by ddevault** [comment]
Thanks @asvetlov and @warsaw!

**Comment by warsaw** [comment]
Yay!  Glad to help.

**Comment by asvetlov** [comment]
Thanks for not the silent disagree but opening a productive discussion!

**Comment by pepoluan** [comment]
Now that we're over the thing-that-shall-not-be-named, let's return to the PR proper 😉 

I'll pick this up.

**Comment by pepoluan** [comment]
**Temporarily** converting this to Draft. Will UnDraft after I test this on my testing systems.

**Comment by pepoluan** [comment]
Test Progress on my Test Systems:

- [x] Windows 10 (via PyCharm tox runner): `(ALL)`
- [x] Windows 10 (via PSCore 7.1.0): `(ALL)`
- [ ] Windows 10 (via Cygwin):

**This comment is undergoing editing**

File: MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1,3 @@
 graft aiosmtpd
-include .coveragerc LICENSE *.cfg *.ini *.py *.rst *.yml
+include LICENSE *.cfg *.ini *.py *.rst *.yml
```
Comment (pepoluan) [inline MANIFEST.in:2]: `.coveragerc` has been renamed to `coverage.cfg`

Reason is explained in a59fd413

File: tox.ini [hunk 2]
```diff
@@ -30,7 +30,8 @@ deps =
     coverage
     diff_cover
 setenv =
-    cov: COVERAGE_PROCESS_START=.coveragerc
+    cov: COVERAGE_RCFILE=coverage.cfg
+    cov: COVERAGE_PROCESS_START=coverage.cfg
```
Comment (pepoluan) [inline tox.ini:34]: `.coveragerc` has been renamed to `coverage.cfg`

Reason is explained in a59fd413

**Comment by pepoluan** [comment]
[All testing done](https://github.com/aio-libs/aiosmtpd/pull/207#issuecomment-752939487).

Will now UnDraft.

**Review by pepoluan — APPROVED** [review]
I personally have no objections to the code / changes. But it will be cheating if I squamerge based on just this approval 😅

**Comment by pepoluan** [comment]
If no objections within 2x24 I'll squamerge this 😄

**Pull request merged**